server/src/models/group.rs

```diff
@@ -0,0 +1,41 @@
+use chrono::prelude::*;
+use diesel::prelude::*;
+
+use serde::{Deserialize, Serialize};
+
+use crate::schema::{groups, user_groups};
+
+pub enum UserGroupRole {
+    User,
```

Comment:
You are not a User of a group, you are a Member of a group. Semantically member is better imho.